doc/rtd/development/module_creation.rst

```diff
@@ -128,3 +128,65 @@ in the ``cloud_final_modules`` section before the ``final-message`` module.
 .. _cloud_init_modules: https://github.com/canonical/cloud-init/blob/b4746b6aed7660510071395e70b2d6233fbdc3ab/config/cloud.cfg.tmpl#L70
 .. _cloud_config_modules: https://github.com/canonical/cloud-init/blob/b4746b6aed7660510071395e70b2d6233fbdc3ab/config/cloud.cfg.tmpl#L101
 .. _cloud_final_modules: https://github.com/canonical/cloud-init/blob/b4746b6aed7660510071395e70b2d6233fbdc3ab/config/cloud.cfg.tmpl#L144
+
```

Comment:
Taking into account:

1. The old signature does not offer any advantage over the new one.
2. Existing external modules already know the old signature.
 
Maybe it is better to not advertise the old signature and remove this whole block from the docs.

Thoughts?

Reply:
Since we support it in the code, I thought it would just put a note on it. I could remove the example, and just leave a note for anyone with questions; when we later drop the support, we can remove the documentation as well.

Reply:
Yes, my way of reasoning is: as the log parameter is a feature that we don't want users to use and they can see the documentation of older version, we should probably not document it and simplify the docs.

I think the example should be removed and I'll leave it open to you to decide if a note should be included.

Reply:
+1 to removing the log parameter from the docs for the reasons @aciba90 mentioned.

Reply:
Updated.